### PR TITLE
Move LoginScreen in own file, fix importing backups

### LINKED ISSUE #506: Import backup not working
* export backup from android (old app)
* login on desktop
* import backup
* `Cannot import backups to accounts in use.` error in console

How is importing a backup supposed to work? Since you need to be logged in in order to select `Import Backup...` how can this have worked ever?

### LINKED ISSUE #763: no option to import a backup as first action after installation
There were 2 bugs related to this issue, but they are closed without visible change in behaviour. That is why create the bug again ..

- Platform (linux/mac/windows): linux
- Delta Chat Desktop Version: 0.102.0 preview
- Expected behavior: after installation the first screen should have an option to load a backup (from other device) BEFORE I insert the credentials.  Loading the export would prevent user from more complex settings (my settings will fail in Desktop while Android versions worked fine ?!?!?!)
- Actual behavior: I MUST add my credentials and then I might try to load a backup which typically is rejected (see other bugs here).
- Steps to reproduce the problem: export backup, copy to desktop node, install, start DC
- Screenshots:
- Logs:

### LINKED ISSUE #815: DS 0.103.0 crashes on importing backup
- Platform (linux/mac/windows): Ubuntu Budgie 19.04
- Delta Chat Desktop Version: used Jikstras rpgp build from here: https://github.com/deltachat/deltachat-desktop/issues/806#issuecomment-496122040  (non flatpak)
- Expected behavior: import backup flawlessly
- Actual behavior: DS crashes instantly on hitting "open" when selected the DS bak file.
- Steps to reproduce the problem: backuped from DS 0.102.0 (latest flatpak version)
- Screenshots:
- Logs: 
"2019-05-27T09:09:01.435Z"      "main/index"    "INFO"  []      "cwd /home/xxx/.config/DeltaChat"
"2019-05-27T09:09:01.484Z"      "main/menu"     "INFO"  []      "rebuilding menu with locale en"
"2019-05-27T09:09:04.474Z"      "main/state"    "INFO"  []      "Saving state to /home/xxx/.config/DeltaChat/config.json"
"2019-05-27T09:09:10.829Z"      "main/menu"     "INFO"  []      "rebuilding menu with locale de"
"2019-05-27T09:09:11.787Z"      "main/state"    "INFO"  []      "Saving state to /home/xxx/.config/DeltaChat/config.json"
"2019-05-27T09:09:44.201Z"      "main/deltachat/login"  "INFO"  []      "Using deltachat instance /home/xxx/.config/DeltaChat/6f647973736575735f68656c645f6465725f616e74696b65407765622e6465"
"2019-05-27T09:09:48.235Z"      "main/deltachat"        "WARNING"       [{"columnNumber":11,"lineNumber":114,"fileName":"/opt/DeltaChat/resources/app.asar/src/main/deltachat/index.js","functionName":"DeltaChat.dc$
"2019-05-27T09:09:48.236Z"      "main/deltachat"        "WARNING"       [{"columnNumber":11,"lineNumber":114,"fileName":"/opt/DeltaChat/resources/app.asar/src/main/deltachat/index.js","functionName":"DeltaChat.dc$
"2019-05-27T09:09:48.237Z"      "main/deltachat"        "WARNING"       [{"columnNumber":11,"lineNumber":114,"fileName":"/opt/DeltaChat/resources/app.asar/src/main/deltachat/index.js","functionName":"DeltaChat.dc$
"2019-05-27T09:09:48.237Z"      "main/deltachat"        "WARNING"       [{"columnNumber":11,"lineNumber":114,"fileName":"/opt/DeltaChat/resources/app.asar/src/main/deltachat/index.js","functionName":"DeltaChat.dc$
"2019-05-27T09:09:48.238Z"      "main/deltachat"        "WARNING"       [{"columnNumber":11,"lineNumber":114,"fileName":"/opt/DeltaChat/resources/app.asar/src/main/deltachat/index.js","functionName":"DeltaChat.dc$
"2019-05-27T09:09:48.239Z"      "main/deltachat"        "WARNING"       [{"columnNumber":11,"lineNumber":114,"fileName":"/opt/DeltaChat/resources/app.asar/src/main/deltachat/index.js","functionName":"DeltaChat.dc$
"2019-05-27T09:10:03.502Z"      "main/deltachat/login"  "INFO"  []      "Ready"
"2019-05-27T09:10:03.505Z"      "main/deltachat/login"  "INFO"  []      "dc_get_info"   {"deltachat_core_version":"v0.43.0","sqlite_version":"3.27.2","sqlite_thread_safe":"1","libetpan_version":"1.8","openssl_ver$
"2019-05-27T09:10:04.502Z"      "main/state"    "INFO"  []      "Saving state to /home/xxx/.config/DeltaChat/config.json"
"2019-05-27T09:10:07.714Z"      "main/state"    "INFO"  []      "Saving state to /home/xxx/.config/DeltaChat/config.json"
"2019-05-27T09:10:21.690Z"      "main/index"    "ERROR" [{"columnNumber":7,"lineNumber":27,"fileName":"/opt/DeltaChat/resources/app.asar/src/main/index.js","functionName":"process.on","source":"    at process.on $
"2019-05-27T09:10:30.439Z"	"main/index"	"INFO"	[]	"cwd /home/masterb/.config/DeltaChat"
"2019-05-27T09:10:30.447Z"	"main/deltachat/login"	"INFO"	[]	"Using deltachat instance /home/masterb/.config/DeltaChat/6f647973736575735f68656c645f6465725f616e74696b65407765622e6465"
"2019-05-27T09:10:30.481Z"	"main/menu"	"INFO"	[]	"rebuilding menu with locale de"
"2019-05-27T09:10:30.526Z"	"main/deltachat/login"	"INFO"	[]	"Ready"
"2019-05-27T09:10:30.531Z"	"main/deltachat/login"	"INFO"	[]	"dc_get_info"	{"deltachat_core_version":"v0.43.0","sqlite_version":"3.27.2","sqlite_thread_safe":"1","libetpan_version":"1.8","openssl_version":"1.1.1b","rpgp_enabled":"1","compile_date":"May 22 2019, 23:51:10","arch":"64","number_of_chats":"0","number_of_chat_messages":"0","messages_in_contact_requests":"0","number_of_contacts":"0","database_dir":"/home/masterb/.config/DeltaChat/6f647973736575735f68656c645f6465725f616e74696b65407765622e6465/db.sqlite","database_version":"55","blobdir":"/home/xxx/.config/DeltaChat/6f647973736575735f68656c645f6465725f616e74696b65407765622e6465/db.sqlite-blobs","display_name":"0","is_configured":"1","entered_account_settings":"xxxx@web.de 0:***:0:0 0:0:0:0 0","used_account_settings":"xxx@web.de xxx:***:imap.web.de:993 xxx:***:smtp.web.de:587 AUTH_NORMAL IMAP_SSL SMTP_STARTTLS","inbox_watch":"1","sentbox_watch":"1","mvbox_watch":"1","mvbox_move":"1","folders_configured":"3","configured_sentbox_folder":"Gesendet","configured_mvbox_folder":"DeltaChat","mdns_enabled":"0","e2ee_enabled":"1","private_key_count":"1","public_key_count":"0","fingerprint":"0BFA3E444E2993410AD6FFDFC2D0AB7470035656"}
"2019-05-27T09:10:32.506Z"	"main/state"	"INFO"	[]	"Saving state to /home/masterb/.config/DeltaChat/config.json"
"2019-05-27T09:10:33.522Z"	"main/state"	"INFO"	[]	"Saving state to /home/masterb/.config/DeltaChat/config.json"
"2019-05-27T09:10:45.126Z"	"main/index"	"ERROR"	[{"columnNumber":7,"lineNumber":27,"fileName":"/opt/DeltaChat/resources/app.asar/src/main/index.js","functionName":"process.on","source":"    at process.on (/opt/DeltaChat/resources/app.asar/src/main/index.js:27:7)"},{"columnNumber":15,"lineNumber":187,"fileName":"events.js","functionName":"process.emit","source":"    at process.emit (events.js:187:15)"},{"columnNumber":27,"lineNumber":488,"fileName":"internal/bootstrap/node.js","functionName":"process._fatalException","source":"    at process._fatalException (internal/bootstrap/node.js:488:27)"}]	"uncaughtError"	{"message":"dc.backupImport is not a function","stack":"TypeError: dc.backupImport is not a function\n    at EventEmitter.ipcMain.on (/opt/DeltaChat/resources/app.asar/src/main/ipc.js:168:50)\n    at EventEmitter.emit (events.js:182:13)\n    at WebContents.<anonymous> (/opt/DeltaChat/resources/electron.asar/browser/api/web-contents.js:346:13)\n    at WebContents.emit (events.js:182:13)"}

### PR DESCRIPTION
**Pull request by Jikstra**
Missing to complete:
- Updates deltachat-node to v1.0.0-alpha.5
- Moves LoginScreen in own file
- Fixes mounting all dialogs all the time
- Implements Importing of Backups on login screen

closes #815 
fixes #506
fixes #763

### CODE REVIEW

File: src/main/deltachat/backup.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+    dc.importExport(C.DC_IMEX_IMPORT_BACKUP, file)
+  })
+  dc.on('ALL', (event, ...args) => {
+    console.log('backupImport:', event, ...args)
+  })
+  dc.on('2051', (progress) => {
+    this.emit('DC_EVENT_IMEX_PROGRESS', progress)
+    if(progress == '1000') {
+      let addr = dc.getConfig('addr')
+      this.emit('DD_EVENT_BACKUP_IMPORTED', addr)
```
Comment (Jikstra) [inline src/main/deltachat/backup.js]: This is the tricky part, we need to spawn a new and empty dc context, start importing the backup in there, than figure out what the email address is and than move the folder to the right folder. If something goes wrong we also need to remove that temporary folder. @r10s is there maybe a way that we can get a method to extract the address out of a backup, without the need of a context? This way we could create the context already at the correct path which would make things easier.

Comment (Jikstra) [reply src/main/deltachat/backup.js]: Also in my tests yesterday somehow the DC_EVEN_IMEX_PROGRESS event didn't get triggered correctly. Need to investigate this further.

**Comment by Jikstra** [comment]
@ralphtheninja @r10s i investigated the dc close problem a bit further. If I do the dc close call and i still have jobs to do (which is highly possible as we're importing a backup, very likely that there are new emails) it will try to fetch them and do stuff. So it seems like the dcn_close call isn't properly killing the threads. Maybe a -node issue, not sure yet. So current workaround is to disable internet ^^

**Comment by ralphtheninja** [comment]
> So current workaround is to disable internet ^^

:laughing:

**Comment by nicodh** [comment]
What exactly is the problem here? @Jikstra  Can you describe a bit what happens? What is the problem with dc_close? If you share your insights maybe I can continue working on this? Or do you need some core fix and wait for that?

**Comment by Jikstra** [comment]
> What exactly is the problem here? @Jikstra Can you describe a bit what happens? What is the problem with dc_close? If you share your insights maybe I can continue working on this? Or do you need some core fix and wait for that?

The dc close call hangs if there are still imap jobs in the queue. And when importing a backup where new messages arrived, this happens almost always. But this shouldn't block us from merging this, i just need to fix some standard stuff here.

File: src/main/deltachat/backup.js [hunk 2]
```diff
@@ -1,9 +1,103 @@
...
+    log.debug('backup event:', eventStr, data1, data2)
+    dcnEvent.emit(eventStr, data1, data2)
+  })
+
+  dcnEvent.on('DC_EVENT_IMEX_PROGRESS', progress => {
+    this.sendToRenderer('DD_EVENT_IMPORT_PROGRESS', progress / 2)
+    if (progress === 0) {
+      onError('UNKNOWN_ERROR')
+    }
+    if (progress === 1000) {
```
Comment (ralphtheninja) [inline src/main/deltachat/backup.js]: else if

File: src/renderer/components/dialogs/index.js [hunk 3]
```diff
@@ -61,6 +61,7 @@ class Controller extends React.Component {
   render () {
     const { saved, userFeedback, deltachat } = this.props
     const { dialogs } = this.state
+    // ToDo: This is shit. We can't alway renders all Dialogs and show them if something happens. We need to hook them up if they are needed, not always
```
Comment (ralphtheninja) [inline src/renderer/components/dialogs/index.js]: Maybe just do 

```js
return props.isOpen ? <dialog.Component {...props} /> : null
```

https://github.com/deltachat/deltachat-desktop/blob/3f2b6ed55e84c4bfcc23bc5fa0ebb0fd047a6da2/src/renderer/components/dialogs/index.js#L80

File: src/renderer/components/dialogs/index.js [hunk 4]
```diff
@@ -66,14 +66,19 @@ class Controller extends React.Component {
     return (
       <div>
         {Object.values(dialogs).map((dialog) => {
+          let isOpen = dialog.props !== false
+          if (!isOpen) return null
```
Comment (ralphtheninja) [inline src/renderer/components/dialogs/index.js]: :+1:

Comment (ralphtheninja) [reply src/renderer/components/dialogs/index.js]: Please remove the comment as well.

**Comment by Jikstra** [comment]
@ralphtheninja @nicodh this is now ready to test/review

**Comment by nicodh** [comment]
Hmm, my first test (on linux) failed. I exported a back and reimported it immediately. Somehow it didn't get to the point where I was asked if I want to overwrite an existing account, but after some time it crashed.

Her is my log (starting a bit before I started the import)

main/deltachat DEBUG [] Core Event DC_EVENT_INFO  INBOX-IDLE will not be started because of waiting jobs.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  INBOX-jobs started...
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  INBOX-job #9, action 910 started...
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Suspending SENTBOX-thread.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Interrupting SENTBOX-IDLE...
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  SENTBOX-IDLE ended.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Suspending MVBOX-thread.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Interrupting MVBOX-IDLE...
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  IMAP-IDLE interrupted
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  MVBOX-IDLE ended.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  SENTBOX-IDLE will not be started as it was interrupted while not ideling.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Import/export process started.
main/deltachat DEBUG [] Core Event DC_EVENT_IMEX_PROGRESS 10 0
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  MVBOX-IDLE will not be started as it was interrupted while not ideling.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Start housekeeping...
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  7 files in use.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Housekeeping done.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Database closed.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Backup "/home/ndh/.config/DeltaChat/64656c74617465737432406e64682e6d65/db.sqlite" to "/home/ndh/Downloads/DC-Backups/delta-chat-2019-07-20.bak".
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Opened "/home/ndh/.config/DeltaChat/64656c74617465737432406e64682e6d65/db.sqlite".
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Opened "/home/ndh/Downloads/DC-Backups/delta-chat-2019-07-20.bak".
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  EXPORT: total_files_cnt=2
main/deltachat DEBUG [] Core Event DC_EVENT_IMEX_PROGRESS 500 0
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  EXPORTing filename=image.jpeg
main/deltachat DEBUG [] Core Event DC_EVENT_IMEX_PROGRESS 990 0
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  EXPORTing filename=BBH_gravitational_lensing_of_gw150914.webm
main/deltachat DEBUG [] Core Event DC_EVENT_IMEX_FILE_WRITTEN /home/ndh/Downloads/DC-Backups/delta-chat-2019-07-20.bak 0
main/ipc DEBUG [] DC_EVENT_IMEX_FILE_WRITTEN: /home/ndh/Downloads/DC-Backups/delta-chat-2019-07-20.bak
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Database closed.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Import/export completed.
main/deltachat DEBUG [] Core Event DC_EVENT_IMEX_PROGRESS 1000 0
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Unsuspending SENTBOX-thread.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  Unsuspending MVBOX-thread.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  INBOX-jobs ended.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  INBOX-fetch started...
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  MVBOX-fetch started...
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  0 mails read from "INBOX".
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  INBOX-fetch done in 21 ms.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  INBOX-IDLE started...
main/deltachat DEBUG [] Core Event DC_EVENT_IMAP_CONNECTED  IMAP-LOGIN as deltatest2@foo-bar.me ok
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  IMAP-capabilities: LITERAL+ ENABLE UNSELECT ESORT SORT=DISPLAY MULTIAPPEND UIDPLUS ESEARCH LIST-EXTENDED SPECIAL-USE CONDSTORE IDLE CONTEXT=SEARCH URL-PARTIAL CATENATE CHILDREN WITHIN LIST-STATUS QUOTA I18NLEVEL=1 BINARY SASL-IR ID LOGIN-REFERRALS THREAD=REFS THREAD=REFERENCES SEARCHRES THREAD=ORDEREDSUBJECT SNIPPET=FUZZY IMAP4rev1 NAMESPACE COMPRESS=DEFLATE MOVE SORT QRESYNC
failed to close connection: Bad("No mailbox selected (0.001 + 0.000 secs).")
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  IMAP disconnected.
main/deltachat DEBUG [] Core Event DC_EVENT_IMAP_CONNECTED  IMAP-LOGIN as deltatest2@foo-bar.me ok
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  IMAP-capabilities: IMAP4rev1 SEARCHRES LIST-EXTENDED QUOTA LIST-STATUS ENABLE UIDPLUS SPECIAL-USE COMPRESS=DEFLATE ESEARCH ESORT I18NLEVEL=1 LITERAL+ ID CONTEXT=SEARCH QRESYNC LOGIN-REFERRALS URL-PARTIAL THREAD=REFERENCES SORT=DISPLAY MOVE SNIPPET=FUZZY MULTIAPPEND CATENATE SORT SASL-IR IDLE WITHIN THREAD=REFS UNSELECT THREAD=ORDEREDSUBJECT CHILDREN NAMESPACE CONDSTORE BINARY
failed to close connection: Bad("No mailbox selected (0.001 + 0.000 secs).")
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  IMAP disconnected.
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  SENTBOX-IDLE started...
main/deltachat DEBUG [] Core Event DC_EVENT_INFO  IMAP-fake-IDLEing...
main/deltachat/login INFO [] Logged out
main/ipc DEBUG [] RENDER
main/state INFO [] Saving state to /home/ndh/.config/DeltaChat/config.json
Gtk-Message: 22:03:42.855: GtkDialog mapped without a transient parent. This is discouraged.
main/deltachat/backup DEBUG [] Creating dummy dc config for importing at /tmp/4e6a8585c65ba3cd7e851f7beb20e218
main/deltachat/backup DEBUG [] backup event: DC_EVENT_INFO 0 First time init: creating tables in "/tmp/4e6a8585c65ba3cd7e851f7beb20e218/db.sqlite".
main/deltachat/backup DEBUG [] openend context
main/deltachat/backup DEBUG [] Starting backup import of /home/ndh/Downloads/DC-Backups/delta-chat-2019-07-20.bak
main/index ERROR [ { columnNumber: 7,
    lineNumber: 27,
    fileName: '/home/ndh/projekte/deltachat/desktop/src/main/index.js',
    functionName: 'process.on',
    source:
     '    at process.on (/home/ndh/projekte/deltachat/desktop/src/main/index.js:27:7)' },
  { columnNumber: 15,
    lineNumber: 187,
    fileName: 'events.js',
    functionName: 'process.emit',
    source: '    at process.emit (events.js:187:15)' },
  { columnNumber: 27,
    lineNumber: 488,
    fileName: 'internal/bootstrap/node.js',
    functionName: 'process._fatalException',
    source:
     '    at process._fatalException (internal/bootstrap/node.js:488:27)' } ] uncaughtError { message: 'binding.dcn_perform_imap_jobs is not a function',
  stack:
   'TypeError: binding.dcn_perform_imap_jobs is not a function\n    at binding.dcn_open.err (/home/ndh/projekte/deltachat/desktop/src/main/deltachat/backup.js:95:13)' }
/home/ndh/projekte/deltachat/desktop/src/main/index.js:28
  throw err
  ^

TypeError: binding.dcn_perform_imap_jobs is not a function
    at binding.dcn_open.err (/home/ndh/projekte/deltachat/desktop/src/main/deltachat/backup.js:95:13)
npm ERR! code ELIFECYCLE
npm ERR! errno 7
npm ERR! deltachat-desktop@0.105.0-pre2 start: `electron . --debug --log-debug --log-to-console --disable-http-cache`
npm ERR! Exit status 7
npm ERR! 
npm ERR! Failed at the deltachat-desktop@0.105.0-pre2 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ndh/.npm/_logs/2019-07-20T20_05_18_751Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 7
npm ERR! deltachat-desktop@0.105.0-pre2 dev: `npm run build && npm run start`
npm ERR! Exit status 7
npm ERR! 
npm ERR! Failed at the deltachat-desktop@0.105.0-pre2 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:

**Comment by nicodh** [comment]
Ahh -  I see -I have to use the new node bindings...

**Comment by Jikstra** [comment]
Ayay seems like desktop integration tests fail because of core-rust...

**Comment by Jikstra** [comment]
@nicodh @ralphtheninja @Simon-Laux can any of you review this? Should be done now.

File: src/main/deltachat/index.js [hunk 5]
```diff
@@ -61,6 +61,7 @@ class DeltaChatController extends EventEmitter {
    */
   sendToRenderer (eventType, payload) {
     log.debug('sendToRenderer: ' + eventType)
+    windows.main.send('ALL', eventType, payload)
```
Comment (Simon-Laux) [inline src/main/deltachat/index.js]: why send an all event? ... Oh its for debugging.. Can you check if the loglevel is debug before sending it? Or does it have no performance impact at all?

File: src/renderer/components/Login-Styles.js [hunk 6]
```diff
@@ -4,9 +4,30 @@ import {
...
   FormGroup,
-  Intent
+  Intent,
+  ProgressBar
 } from '@blueprintjs/core'
 
+export const DeltaBlueButton = styled.div`
+  -webkit-appearance: button-bevel;
+  background-color: transparent;
+  color: ${props => props.theme.deltaFocusBlue};;
```
Comment (Simon-Laux) [inline src/renderer/components/Login-Styles.js]: after #911 the variables will have different names so we need to remember to change this line when merging #911

Comment (Jikstra) [reply src/renderer/components/Login-Styles.js]: good to know, lets add this as an todo to #911

File: src/main/deltachat/index.js (hunk 5 above)
Comment (Jikstra) [reply src/main/deltachat/index.js]: Don't think it will affect performance in a meaningful way

**Pull request merged**